### DISCUSSION TITLE
gRPC not reporting

### DISCUSSION THREAD

**Discussion by mikeumus** [jaegertracing/jaeger#2836] [Q&A] [answered]
```
  docker run -d \
    -p6831:6831/udp \
    -p6832:6832/udp \
    -p2052:2052 \
    -p2053:2053 \
    jaegertracing/all-in-one:latest \
    --collector.http-server.host-port=:2053 \
    --query.http-server.host-port=:2052 \
    --collector.grpc-server.host-port=:2082 \
    --reporter.type=grpc \
    --reporter.grpc.host-port=ingest.lightstep.com:443
```

Specifically the `--collector.grpc-server.host-port=:2082`

Does that mean I listen with `grpcdump` on port `2082`?
`sudo grpcdump -i lo -p 2082`

I don't see any activity when I send traces to the Jaeger server with that. 
Is it supposed to report per trace sent to the `jaeger/all-in-one` collector?

**Comment by yurishkuro** [comment]
How are you generating traces, with what SDKs?  It looks like the agent is configured to send data to lightstep, so if you're using standard Jaeger SDK with UDP, then nothing would be reaching collector port 2082.

**Reply by mikeumus** [reply]
I'm using the `optelemetry-rust` crate to send traces via TCP to:

```
https://trace.filevacuum.com:2053/api/traces
```

**Reply by yurishkuro** [reply]
I suspect the issue is with your networking setup, since clearly there's some kind of reverse proxy you're using, so tcpdump showing nothing on the collector port just indicates that that data is getting lost upstream. I am not familiar with Rust SDK, but if it's using either gRPC or HTTP to submit data to Jaeger, it should be receiving responses from the server in case of successful submission. In your case I would expect some kinds of errors on the client side.

**Comment by mikeumus** [comment, marked as answer]
Don't forget to specify all the ports you need when running the docker image:

```
  docker run -d \
    -p6831:6831/udp \
    -p6832:6832/udp \
    -p2052:2052 \
    -p2053:2053 \
    -p 2082:2082 \
    -p 2087:2087 \
    jaegertracing/all-in-one:latest
```
